jpa-criteria/impl/src/main/java/com/blazebit/persistence/criteria/impl/InternalQuery.java

```diff
@@ -591,14 +650,91 @@ private FromBuilder<?> renderJoins(FromBuilder<?> cb, SubqueryInitiator<?> subqu
                 }
             }
 
-            renderJoins(cb, null, fetching, context, alias, (Set<BlazeJoin<?, ?>>) (Set<?>) j.getBlazeJoins());
+            renderJoins(cb, (SubqueryInitiator<?>) null, fetching, context, alias, (Set<BlazeJoin<?, ?>>) (Set<?>) j.getBlazeJoins());
+
+            Collection<TreatedPath<?>> treatedPaths = (Collection<TreatedPath<?>>) (Collection) join.getTreatedPaths();
+            if (treatedPaths != null && treatedPaths.size() > 0) {
+                for (TreatedPath<?> treatedPath : treatedPaths) {
+                    AbstractJoin<?, ?> treatedJoin = (AbstractJoin<?, ?>) treatedPath;
+                    String treatedParentPath = "TREAT(" + alias + " AS " + treatedPath.getTreatType().getName() + ')';
+                    renderJoins(cb, (SubqueryInitiator<?>) null, fetching, context, treatedParentPath, (Set<BlazeJoin<?, ?>>) (Set<?>) treatedJoin.getBlazeJoins());
+                }
+            }
+        }
+
+        return cb;
+    }
+
+    private FromBuilder<?> renderJoins(FromBuilder<?> cb, LeafOngoingSetOperationSubqueryBuilder<?> subqueryInitiator, boolean fetching, RenderContextImpl context, String parentPath, Set<BlazeJoin<?, ?>> joins) {
```

Comment:
Necessary overload for the same reason as `renderSubqueryFrom(LeafOngoingSetOperationSubqueryBuilder<?> initiator, RenderContextImpl context)`.

Reply:
That's fine.